In [1]:
import pandas as pd
import numpy as np

## This Jupyter notebook loads data downloaded from https://www.kaggle.com/szrlee/stock-time-series-20050101-to-20171231/home and in the thesis we only use subset of data as discussed in https://methods.sagepub.com/dataset/howtoguide/multivariate-hawkes-in-djia-2018#i100

In [2]:
df = pd.read_csv('archive/all_stocks_2006-01-01_to_2018-01-01.csv')

In [3]:
df

Date   Open   High    Low  Close    Volume  Name
0      2006-01-03  77.76  79.35  77.24  79.11   3117200   MMM
1      2006-01-04  79.49  79.49  78.25  78.71   2558000   MMM
2      2006-01-05  78.41  78.65  77.56  77.99   2529500   MMM
3      2006-01-06  78.64  78.90  77.64  78.63   2479500   MMM
4      2006-01-09  78.50  79.83  78.46  79.02   1845600   MMM
...           ...    ...    ...    ...    ...       ...   ...
93607  2017-12-22  71.42  71.87  71.22  71.58  10979165  AABA
93608  2017-12-26  70.94  71.39  69.63  69.86   8542802  AABA
93609  2017-12-27  69.77  70.49  69.69  70.06   6345124  AABA
93610  2017-12-28  70.12  70.32  69.51  69.82   7556877  AABA
93611  2017-12-29  69.79  70.13  69.43  69.85   6613070  AABA

[93612 rows x 7 columns]

In [4]:
df.columns

Index(['Date', 'Open', 'High', 'Low', 'Close', 'Volume', 'Name'], dtype='object')

In [5]:
df = df[['Date','Close','Name']] # select subset of the dataframe

In [6]:
ticker = df['Name'].unique() # unique stock symbols
num_of_stocks = np.shape(ticker)[0] # 31 stocks
ticker

array(['MMM', 'AXP', 'AAPL', 'BA', 'CAT', 'CVX', 'CSCO', 'KO', 'DIS',
       'XOM', 'GE', 'GS', 'HD', 'IBM', 'INTC', 'JNJ', 'JPM', 'MCD', 'MRK',
       'MSFT', 'NKE', 'PFE', 'PG', 'TRV', 'UTX', 'UNH', 'VZ', 'WMT',
       'GOOGL', 'AMZN', 'AABA'], dtype=object)

### Use pivot method to rehshape data by index/column values in this case dates/ticker symbols values. Data is reshaped from 93612 observations by 7 columns to 3020 rows by 31 columns.

### Each observation (row) is indexed by a unique date starting from the date 2006/01/03 to 2017-12-29. Row indicates time axis meaning it represents time dimension. An observation (or a given row). The column variable contains information of the daily closing price of the 31 stocks and each column represents a particlar stock. The name of the column variable is the ticker symbol of each stock.

In [7]:
#df_table = pd.pivot_table(df, values='Close', index='Date', columns='Name')
df_table = df.pivot(index='Date', columns= 'Name', values='Close')

In [8]:
df_table

Name         AABA    AAPL     AMZN    AXP      BA     CAT   CSCO     CVX  \
Date                                                                       
2006-01-03  40.91   10.68    47.58  52.58   70.44   57.80  17.45   59.08   
2006-01-04  40.97   10.71    47.25  51.95   71.17   59.27  17.85   58.91   
2006-01-05  41.53   10.63    47.65  52.50   70.33   59.27  18.35   58.19   
2006-01-06  43.21   10.90    47.87  52.68   69.35   60.45  18.77   59.25   
2006-01-09  43.42   10.86    47.08  53.99   68.77   61.55  19.06   58.95   
...           ...     ...      ...    ...     ...     ...    ...     ...   
2017-12-22  71.58  175.01  1168.36  98.74  295.10  155.75  38.55  124.98   
2017-12-26  69.86  170.57  1176.76  98.57  295.36  156.44  38.48  125.98   
2017-12-27  70.06  170.60  1182.26  99.13  295.62  157.52  38.56  125.55   
2017-12-28  69.82  171.08  1186.10  99.70  296.35  158.42  38.59  125.58   
2017-12-29  69.85  169.23  1169.47  99.31  294.91  157.58  38.30  125.19   

Name           DIS     GE  ...   MSFT    NKE    PFE     PG     TRV     UNH  \
Date                       ...                                               
2006-01-03   24.40  35.37  ...  26.84  10.74  23.78  58.78   45.99   61.73   
2006-01-04   23.99  35.32  ...  26.97  10.69  24.55  58.89   46.50   61.88   
2006-01-05   24.41  35.23  ...  26.99  10.76  24.58  58.70   46.95   61.69   
2006-01-06   24.74  35.47  ...  26.91  10.72  24.85  58.64   47.21   62.90   
2006-01-09   25.00  35.38  ...  26.86  10.88  24.85  59.08   47.23   61.40   
...            ...    ...  ...    ...    ...    ...    ...     ...     ...   
2017-12-22  108.67  17.50  ...  85.51  63.29  36.14  92.13  134.39  220.00   
2017-12-26  108.12  17.43  ...  85.40  63.65  36.21  92.48  134.78  219.60   
2017-12-27  107.64  17.38  ...  85.71  62.95  36.33  92.10  134.77  220.42   
2017-12-28  107.77  17.36  ...  85.72  62.95  36.37  92.07  135.66  222.77   
2017-12-29  107.51  17.45  ...  85.54  62.55  36.22  91.88  135.64  220.46   

Name           UTX     VZ    WMT    XOM  
Date                                     
2006-01-03   56.53  30.38  46.23  58.47  
2006-01-04   56.19  31.27  46.32  58.57  
2006-01-05   55.98  31.63  45.69  58.28  
2006-01-06   56.16  31.35  45.88  59.43  
2006-01-09   56.80  31.48  45.71  59.40  
...            ...    ...    ...    ...  
2017-12-22  127.23  53.19  98.21  83.97  
2017-12-26  127.14  53.22  99.16  83.98  
2017-12-27  127.58  53.28  99.26  83.90  
2017-12-28  128.12  53.43  99.40  84.02  
2017-12-29  127.57  52.93  98.75  83.64  

[3020 rows x 31 columns]

In [9]:
df_table.describe()

Name          AABA         AAPL         AMZN          AXP           BA  \
count  3019.000000  3019.000000  3019.000000  3020.000000  3020.000000   
mean     28.412726    64.662892   299.376231    59.385315   101.515868   
std      13.258163    44.536448   279.980161    19.371901    47.480903   
min       8.950000     7.240000    26.070000    10.260000    29.360000   
25%      16.130000    23.420000    81.090000    45.715000    70.600000   
50%      27.100000    57.920000   205.440000    57.815000    85.490000   
75%      36.635000    99.480000   375.140000    75.892500   130.615000   
max      72.930000   176.420000  1195.830000    99.700000   297.900000   

Name           CAT         CSCO          CVX          DIS           GE  ...  \
count  3020.000000  3019.000000  3020.000000  3020.000000  3020.000000  ...   
mean     80.480182    24.039632    94.625255    57.967381    25.122232  ...   
std      20.945808     5.034167    19.426383    31.278997     7.415372  ...   
min      22.170000    13.620000    54.080000    15.590000     6.660000  ...   
25%      68.447500    20.105000    77.737500    32.430000    18.957500  ...   
50%      81.765000    23.890000    97.790000    42.015000    25.405000  ...   
75%      93.547500    27.545000   109.587500    92.007500    30.535000  ...   
max     158.420000    38.740000   134.850000   121.690000    42.120000  ...   

Name          MSFT          NKE          PFE           PG          TRV  \
count  3019.000000  3020.000000  3020.000000  3020.000000  3020.000000   
mean     36.513412    29.916977    25.442477    70.970583    73.373007   
std      14.694656    17.195567     6.511962    11.223018    28.409330   
min      15.150000     9.520000    11.660000    44.180000    30.500000   
25%      26.835000    15.150000    19.367500    62.560000    50.097500   
50%      30.000000    23.535000    25.610000    67.210000    59.385000   
75%      44.400000    47.712500    31.295000    80.942500   100.007500   
max      86.850000    67.160000    37.310000    94.400000   136.360000   

Name           UNH          UTX           VZ          WMT          XOM  
count  3020.000000  3020.000000  3020.000000  3020.000000  3020.000000  
mean     73.568149    84.906086    41.143642    62.280285    81.012185  
std      47.422996    21.546238     7.694002    12.747477    10.397202  
min      16.300000    37.560000    25.080000    42.270000    56.570000  
25%      42.772500    68.000000    34.600000    50.870000    73.230000  
50%      54.455000    79.600000    42.475000    59.030000    82.605000  
75%     102.550000   105.727500    48.120000    73.765000    88.352500  
max     228.170000   128.120000    56.530000    99.620000   104.380000  

[8 rows x 31 columns]

In [10]:
# Export the subsetted and pivoted data 
df_table.to_csv("dataset-djia-2018-subset2.csv", index=True)